### PR TITLE
[battery_platform_interface] Introduce package.

### PR DESCRIPTION
**Pull request by yash1200**
## Description

In this PR, I have added platform interface package to the battery plugin.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by yash1200** [comment]
Hey @ditman kindly have a look at this one.

File: packages/battery/battery_platform_interface/lib/enums/batteryState.dart [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/// Indicates the current battery state.
```
Comment (ditman) [inline packages/battery/battery_platform_interface/lib/enums/batteryState.dart:1]: Please, don't use camelCase in the name of this file, this file should be: `battery_state.dart`

Comment (yash1200) [reply packages/battery/battery_platform_interface/lib/enums/batteryState.dart]: I renamed it now.

**Comment by ditman** [comment]
Tagged and [published](https://pub.dev/packages/battery_platform_interface) `battery_platform_interface: ^1.0.0`

**Comment by yash1200** [comment]
Thank You :)

**Comment by yash1200** [comment]
In porting `battery` to use platform interface I have to delete the `battery/test` now. Can I do that ? Because the tests are no in `battery_platform_interface`.

**Comment by ditman** [comment]
@yash1200 you can leave the tests in `battery/test`. Why do you have to delete them? Because the method channel is not publicly visible anymore?

(Tests in `battery/test` should call methods in the plugin class, and in the platform_interface they should call directly methods on the method_channel class)

**Pull request merged**